staging/src/k8s.io/controller-manager/config/v1/types.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LeaderMigrationConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// LeaderName is the name of the leader election resource that protects the migration
+	// E.g. 1-20-KCM-to-1-21-CCM
+	LeaderName string `json:"leaderName"`
+
```

Comment:
Interestingly while we have removed ResourceLock from V1, it still exists in the internal type and in the v1beta1 and v1alpha1 types. So there is nothing stopping someone from still configuring it.

Reply:
We don't allow users to use internal types in the config files though right? Or at least it is discouraged

Reply:
Our API convention does not allow (de-)serialize from/to internal types. Here we keep this field in internal, alpha, and beta versions. If the user uses it in versions that still supported the now removed values, they can use v1beta1 config, which will be correctly converted into the internal version of config.

Reply:
This all sounds right. The alpha and beta APIs remain available, so the internal plumbing supports all the functionality they support.  Because this is a config file object, and not a resource object, there is no need to support conversion, so there is no need to support the round trip requirements that resource objects must adhere to.

Reply:
Sorry my point in mentioning the internal type was to indicate that the configurable resource functionality was still supported by the code and that as Joe says you can still access that behavior by using the alpha/beta APIs in the config file.